network/src/connectivity_manager/mod.rs

```diff
@@ -483,6 +483,17 @@ where
             self.network_context
         );
 
+        // Log the eligible peers state periodically so we see what our current state is
+        sample!(SampleRate::Duration(Duration::from_secs(60)), {
+            let eligible = self.eligible.read().unwrap().clone();
+            let elig_peers: Vec<_> = eligible.keys().collect();
+            info!(
+                NetworkSchema::new(&self.network_context),
+                peers = ?elig_peers,
+                "Current eligible peers"
```

Comment:
why not print the values too? I think we expect this to be pretty static but the network addresses could change

Reply:
Yeah, we could stick the addresses in too.

Reply:
the source of truth is pretty much `self.peer_addrs`, which covers the peer ids, the network addresses, and the public keys (inside the network addresses). so we could probably just dump that whole thing.

Reply:
So, now I'm confused.  Why do we have `peer_keys` and `eligible` if they're the same but one has a lock?  Does one just not include the self address?

Yeah, I can change it to just dump the network addresses.

Reply:
I've changed it to print `peer_keys` and `peer_addrs` see overview